### DISCUSSION TITLE
update for TB 91

### DISCUSSION THREAD

**Discussion by opto** [xnotepp/xnote#66] [General]
please try this for TB 91 beta5.

Tested on 64 bit TB on win 10.

thanks, Klaus

<hr /><em>This discussion was created from the release <a href='https://github.com/xnotepp/xnote/releases/tag/v3.2.1'>update for TB 91</a>.</em>

**Comment by mbiba** [comment]
I am seeing PARSE ERROR on each taken note. It occurs after couple of hours using:

![image](https://user-images.githubusercontent.com/888367/130442844-80ed9de7-99f0-4540-982b-9f2887344fab.png)

XML parsing error: No root element found
Address: chrome: //xnote/content/xnote-window.xhtml
Row 1, column 1:

32bit latest TB on latest windows 10 64 bit.

**Reply by jochenpeters** [reply]
For me (Windows 10, 64 Bit, version 2004) the plugin works to attach notes in Thunderbird 91.0.1, but the column XNote in the mail overview/list does not show XNotes and does therefore not allow to sort for mails with XNotes.

It would be great if the column would also work again :-)

**Reply by claudiutamas** [reply]
The same error when I try to create note:
"XML Parsing Error: no root element found
Location: chrome://xnote/content/xnote-window.xhtml
Line Number 1, Column 1:" (and an empty line)

I use Windows 10 x64 20H2, Thunderbird 91.2 (32b)

**Comment by skaleta** [comment]
For me is the some problem as jochenpeters:
- TB 91 -> New Profile -> … xNote working very good, not problems
- TB 91 -> open current profile (I use on TB 78) -> xNote instalation for TB 91 is not problem, but is not a column in the mesage list panel and Xnote is not in the column selection. But new note or edit note is working. My notes I can see in the body of the e-mail.
![tb91 xnote problem](https://user-images.githubusercontent.com/117975/130949871-ebe7d7cf-b31f-4391-85e3-1a8e276881f7.jpg)

**Reply by jochenpeters** [reply]
I also use my previous TB78 profile. I can add the XNote column, but adding XNotes to mails does not mark them in this column. The mail displays the note, but I have to know which mail contains a note - sorting (like for starred messeages) for XNote'd is not possible.

**Reply by jochenpeters** [reply]
After updating to TB 91.0.3, the AddOn stopped working. Removing and re-installing it, it worked again (once even with the XNote column), but stopped working after closing and restarting TB. Now, I cannot even select the XNote column any more - and also not add notes (or see exiting notes).

**Reply by mbiba** [reply]
Can confirm that reinstalling is only temporarilly helpful. Sometimes it works for a while and suddenly stops working after couple of minutes...

**Reply by jochenpeters** [reply]
Side question (not related to XNote but TH91):

Do you also note that closing TB 91.0.1 or 91.0.3 does not really terminate all processes ? Restarting TB after closing it,I always get a popup that it is still running with some "Close it" button - and after pressing that button, TB restarts ?

**Reply by mbiba** [reply]
No processes running on my PC after closing thunderbird...

**Reply by jochenpeters** [reply]
And restarting TB is fine ? (In the task manager, I also see nothing)

**Reply by mbiba** [reply]
How to restart TB? After the restart of my PC I cant edit or see note. Existing note is only visible in mail preview. Column works well. Restarting PC has no impact on this behaviour.

**Comment by RealSchwappi** [comment]
Is there Anything new for this problem?
Since yesterday, i have the same Problem:

After installing the 3.2.1-Version and updating TB 91.2.1 (32-Bit) the same message in the notes-box appears:
_XML-Verarbeitungsfehler: Kein Wurzel-Element gefunden
Adresse: chrome://xnote/content/xnote-window.xhtml
Zeile Nr. 1, Spalte 1:_

![2021-10-28 09_45_09-xnote - Mozilla Thunderbird](https://user-images.githubusercontent.com/93313304/139216781-55f4726a-c637-4871-b26b-12849e9e72c0.png)

This problem appears when selecting an message with notes or trying to create a new note.
Bottom-Message (the upper line of the Mail) is o.k. and displayed the whole time.

After reinstalling the 321 everything is o.k. until I restart TB again.

Its not so comfortable to install the Notes-xpi after every start of TB ;-)

**Comment by opto** [comment]
does 3.2.8 help for you?

https://github.com/xnotepp/xnote/releases/tag/v3.2.8

**Reply by RealSchwappi** [reply]
Great. It works great again :-)
But why does tb this update not automaticity? 

fyi: I use XNote since many years with more than 1000 notes in different mail accounts with cloud based storage-file (dropbox) on different computers. thx for this addon :-)

**Reply by jochenpeters** [reply]
Unfortunately, for me it still fails - it works once immediately after installation but stops working after a TB restart.
In addition, as before, it seems to prevent that Thunderbird can terminate all processes when I close it - crash reports refer to the AddOn.
Sorry ...

**Reply by jochenpeters** [reply]
After creating a new profile, the AddOn works fine - something must have been corrupt in the profile taht worked well until TB 78.x.

**Comment by jochenpeters** [comment]
**I have a new question:**

Does XNote++ allow to filter mails by the note text ?
That would be very convenient to add keywords to mails that can later be searched :-)

**Comment by EliotStearns** [comment]
I also still have the problem with "no root element found", but I found a workaround: The problem does not occur when I add a new note (to any email) after starting Thunderbird, before opening any email that has a note already. After that, also existing notes open without a problem. There does not need to be any text, I can also delete the new note immediately again - it seems that as soon as the note window is initialized once, everything's fine.
Also, many thanks for this add-on from me as well, it's the most important one of them all for me, couldn't live without it! :)

**Comment by opto** [comment]
thanks for that observation.  I will check for the difference between note creating and note opening.
